Feature/spring upgrade

trapnine opened the pull request.

Commit:
use jdk 1.8

Commit:
Update for dependency tree fixes.

Commit:
fix formatting

millerts commented:
@trapnine and I discussed this offline.  For now we're going to leave this PR out here until we can work through some of the downstream projects handling the upgrade problems.  Once we're confident that it won't be too painful we'll start merging stuff

millerts committed:
Add spring data mongo to the dependency list

millerts committed:
Add spring data redis, spring context support, and spring jdbc to the pom specifying the proper versions

Commit:
bump jackson and add spring-web

millerts committed:
Merge pull request #9 from SmarterApp/origin/feature/spring-upgrade-jj

bump jackson and add spring-web

millerts committed:
Add spring security oath and spring security core

millerts committed:
Merge branch 'feature/spring-upgrade' of github.com:SmarterApp/SS_SharedBuild into feature/spring-upgrade

millerts committed:
Added the spring bean dependency declaration setting the version

Commit:
[maven-release-plugin] prepare release 4.0.0.RELEASE

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
upgrade mongodb

Commit:
manage intra-project versions in single location

added the following projects to the pom’s dependency management section:
shared-code, prog-mgmnt-client, prog-mgmnt-client-null-impl
